### PR TITLE
MTM-56104 add info about details of device counting for usage statistics

### PR DESCRIPTION
**Pull request by jkot-softwareag**
I am not sure if this is not too detailed for [this page](https://cumulocity.com/docs/enterprise-tenant/usage-and-billing/) of documentation, but from the other hand one of customers understood current explanation incorrectly and we had a long discussion in the ticket.

### CODE REVIEW

**Review by martin-sag — APPROVED** [review]

**Review by d-mariak — APPROVED** [review]

File: content/enterprise-tenant/usage-and-billing.md [hunk 1]
```diff
@@ -75,7 +75,7 @@ The following information is provided for each subtenant (not completely visible
     </tr>
     <tr>
       <td align="left">Root devices</td>
-      <td align="left">Number of root devices, excluding child devices</td>
+      <td align="left">Number of root devices, excluding child devices, see <a href="#devices-count">Devices count</a> for details</td>
```
Comment (BeateRixen) [inline content/enterprise-tenant/usage-and-billing.md:78]:
```suggestion
      <td align="left">Number of root devices excluding child devices, see also <a href="#devices-count">Devices count details</a> </td>
```

File: content/enterprise-tenant/usage-and-billing.md [hunk 2]
```diff
@@ -176,6 +176,15 @@ The date/time range used here might differ from your server time due to differen
 
 The CSV file will be downloaded to your file system.
 
+#### Devices count details {#devices-count}
+For the device count calculations we use backwards compatibility approach and assume that only the top level devices are marked with the [device marker](/device-integration/fragment-library/#device-marker) fragment (`c8y_IsDevice`).
+For this reason, the following formulas are used in the calculations:
```
Comment (BeateRixen) [inline content/enterprise-tenant/usage-and-billing.md:181]:
```suggestion
Accordingly, the following formulas are used in the calculations:
```

File: content/enterprise-tenant/usage-and-billing.md [hunk 3]
```diff
@@ -176,6 +176,15 @@ The date/time range used here might differ from your server time due to differen
...
+* root devices - all devices with the `c8y_IsDevice` fragment
+* all devices - all devices with the `c8y_IsDevice` fragment and their children from the whole device hierarchy
+* leaf devices - only leafs of the device hierarchies starting from devices with `c8y_IsDevice` fragment
```
Comment (BeateRixen) [inline content/enterprise-tenant/usage-and-billing.md:184]:
```suggestion
* leaf devices - only leafs of the device hierarchies starting from devices with the `c8y_IsDevice` fragment
```

File: content/enterprise-tenant/usage-and-billing.md [hunk 4]
```diff
@@ -176,6 +176,15 @@ The date/time range used here might differ from your server time due to differen
...
+
+In case child devices are also marked with `c8y_IsDevice` fragment, the calculation results may look different than expected.
```
Comment (BeateRixen) [inline content/enterprise-tenant/usage-and-billing.md:186]:
```suggestion
If child devices are also marked with the `c8y_IsDevice` fragment, the calculation results may look different than expected.
```

**Review by karolkaczmarski-sag — APPROVED** [review]

File: content/enterprise-tenant/usage-and-billing.md [hunk 5]
```diff
@@ -176,6 +176,15 @@ The date/time range used here might differ from your server time due to differen
 
 The CSV file will be downloaded to your file system.
 
+#### Devices count details {#devices-count}
+For the device count calculations we use backwards compatibility approach and assume that only the top level devices are marked with the [device marker](/device-integration/fragment-library/#device-marker) fragment (`c8y_IsDevice`).
```
Comment (BeateRixen) [inline content/enterprise-tenant/usage-and-billing.md:180]:
```suggestion
The device count calculations assume that only top-level devices are marked with the [device marker](/device-integration/fragment-library/#device-marker) fragment `c8y_IsDevice`.
```

File: content/change-logs/platform-services/cumulocity-undefined-device-count-in-usage-statistics-detalis-explained.md [hunk 6]
```diff
@@ -0,0 +1,20 @@
...
+    label: Core platform
+build_artifact:
+  - value: tc-QHwMfWtBk7
+    label: cumulocity
+ticket: MTM-56104
+version: 10.20.0.0
+
+---
+
+The details of the algorithms counting root devices, all devices, and leaf devices are explained in the documentation of Usage statistics and billing.
```
Comment (BeateRixen) [inline content/change-logs/platform-services/cumulocity-undefined-device-count-in-usage-statistics-detalis-explained.md:20]:
```suggestion
Details about the algorithms counting root devices, all devices, and leaf devices have been added to the user documentation, see [Usage statistics and billing](/enterprise-tenant/usage-and-billing/). 
```

**Comment by jkot-softwareag** [comment]
@BeateRixen I think we can backport it to few versions. Maybe 2024 and 10.18? It is working like this for years.

**Review by BeateRixen — APPROVED** [review]

**Comment by BeateRixen** [comment]
> @BeateRixen I think we can backport it to few versions. Maybe 2024 and 10.18? It is working like this for years.

@jkot-softwareag Can you please create a PR for 2024 with the respective version number for 2024 in the change log file? For older releases we would also need separate PRs as the documentation has a different structure.

**Comment by jkot-softwareag** [comment]
> > @BeateRixen I think we can backport it to few versions. Maybe 2024 and 10.18? It is working like this for years.
> 
> @jkot-softwareag Can you please create a PR for 2024 with the respective version number for 2024 in the change log file? For older releases we would also need separate PRs as the documentation has a different structure.

Created PRs: 
https://github.com/SoftwareAG/c8y-docs/pull/2060, 
https://github.com/SoftwareAG/c8y-docs/pull/2059

**Pull request merged**